week2/homework/src/index.js

```diff
@@ -1,3 +1,183 @@
+${chalk.inverse.green('Removes a todo, specified by the index.')}
+node . list
+${chalk.inverse.green('Lists all todos.')}
+node . done <index>
+${chalk.inverse.green('Marks a todo as done, specified by the index.')}
+node . update <index> 'new todo'
+${chalk.inverse.green(
+      'Replaces a todo specified by the index withe a new todo.'
+    )}
+node . reset
+${chalk.inverse.green('Resets the todos file.')}
+node . OR node . help
+${chalk.inverse.green('Shows the available commands (this).')}`,
+    {
+      theme: 'round',
+      color: 'green',
+      align: 'left'
+    }
+  )
+};
+
+async function main() {
+  const userIndexInput = parseInt(args[1]);
+  const allTodos = await readJSON();
+
+  function addTodo() {
+    const userTodoInput = args[1];
+    if (userTodoInput === undefined) {
+      console.log(
+        chalk.red('Error while adding a todo, see this:\n'),
+        helpMessages.add
+      );
+    } else {
+      allTodos.push({
+        task: userTodoInput,
+        done: 'not done'
+      });
+      storeTodos(allTodos);
+      console.log(`added todo: '${chalk.cyan(args[1])}'`);
+    }
+  }
+
+  function removeTodo() {
+    if (userIndexInput >= allTodos.length) {
+      console.log(`${chalk.red("didn't find a todo with that index")}`);
+      console.log(helpMessages.remove);
+    }
+    allTodos.forEach((todo, index) => {
+      if (userIndexInput === index) {
+        allTodos.pop(index);
+        storeTodos(allTodos);
+        console.log(`successfully removed '${chalk.cyan(todo.task)}'`);
+      }
+    });
+  }
+
+  function listTodos() {
+    if (allTodos.length === 0) {
+      console.log('no todos yet, start by adding one');
+      console.log(helpMessages.add);
+    } else {
+      console.log('your todos are:');
+      allTodos.forEach((todo, index) => {
+        console.log(
+          `  ${chalk.cyan(index)}- ${todo.task} ${
+            todo.done === 'done'
+              ? chalk.green.bold(todo.done)
+              : chalk.magenta.bold(todo.done)
+          }`
+        );
+      });
+    }
+  }
+
+  function markTodoAsDone() {
+    allTodos.forEach((todo, index) => {
+      if (index === userIndexInput) {
+        if (todo.done === 'done') {
+          console.log(
+            `${chalk.inverse(`you've already marked '${todo.task}' as done`)}`
+          );
+        } else {
+          todo.done = 'done';
+          console.log(`Successfully marked '${chalk.cyan(todo.task)}' as done`);
+          storeTodos(allTodos);
+        }
+      }
+    });
+    if (userIndexInput >= allTodos.length) {
+      console.log(`${chalk.red("didn't find a todo with that index")}`);
+    }
+  }
+
+  function updateTodo() {
+    allTodos.forEach((todo, index) => {
+      if (index === userIndexInput) {
+        const oldTodo = todo.task;
+        console.log(
+          `updated '${chalk.red(oldTodo)}' with '${chalk.cyan(args[2])}'`
+        );
+        allTodos[index].task = args[2];
+        allTodos[index].done = 'not done';
```

Comment:
maybe if I wasn't so blind I could've used **todo.task**, and **todo.done** instead of **allTodos[index].task** and **allTodos[index].done** (lines 147 & 148)